pallets/liquidity-mining/src/lib.rs

```diff
@@ -569,6 +569,59 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		})
 	}
 
+	/// Update global farm's main fields.
+	///
+	///
+	/// Parameters:
+	/// - `global_farm_id`: global farm id.
+	/// - `planned_yielding_periods`: planned number of periods to distribute `total_rewards`.
+	/// - `yield_per_period`: percentage return on `reward_currency` of all pools.
+	/// - `min_deposit`: minimum amount of LP shares to be deposited into liquidity mining by each user.
+	fn update_global_farm(
+		global_farm_id: GlobalFarmId,
+		planned_yielding_periods: PeriodOf<T>,
+		yield_per_period: Perquintill,
+		min_deposit: Balance,
+	) -> Result<(), DispatchError> {
+		ensure!(min_deposit.ge(&MIN_DEPOSIT), Error::<T, I>::InvalidMinDeposit);
+		ensure!(
+			planned_yielding_periods >= T::MinPlannedYieldingPeriods::get(),
+			Error::<T, I>::InvalidPlannedYieldingPeriods
+		);
+		ensure!(!yield_per_period.is_zero(), Error::<T, I>::InvalidYieldPerPeriod);
+
+		<GlobalFarm<T, I>>::try_mutate(global_farm_id, |maybe_global_farm| {
+			let global_farm = maybe_global_farm.as_mut().ok_or(Error::<T, I>::GlobalFarmNotFound)?;
+
+			ensure!(global_farm.state.is_active(), Error::<T, I>::GlobalFarmNotFound);
+
+			//Sync global farm to get right accumulated_rpz and pending rewards
```

Comment:
I don't find this comment helpful, it basically says same thing as doc comment of `sync_global_farm`